File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int count_letters(string txt);
int count_words(string txt);
int count_sents(string txt);

int main(void)
{
    // Prompt user for text string
    string text = get_string("Text: ");

    // Functions which count letters, words, and sentences
    int lets = count_letters(text);
    int wds = count_words(text);
    int sts = count_sents(text);

    // Defining L and S for Coleman-Liau index
    float L = ((float) lets / wds) * 100;
    float S = ((float) sts / wds) * 100;

    // Defining Coleman-Liau index formula for reading comprehension level
    float CLI = 0.0588 * L - 0.296 * S - 15.8;
    // Testing the reading comprehension level against the CLI formula
    if (CLI >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (CLI >= 1)
    {
        printf("Grade %i\n", (int) (CLI + 0.5));
    }
    else
    {
        printf("Before Grade 1\n");
    }
}
// Implementing count letters function
int count_letters(string txt)
{
    int letters = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        if (txt[i] >= 'A' && txt[i] <= 'Z')
        {
            letters++;
        }
        else if (txt[i] >= 'a' && txt[i] <= 'z')
        {
            letters++;
        }
    }
    return letters;
}
// Implementing count words function
int count_words(string txt)
{
    int words = 1;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        if (txt[i] == ' ')
        {
            words++;
        }
    }
    return words;
}
// Implementing count sentences function
int count_sents(string txt)
{
    int sents = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
        {
            sents++;
        }
    }
    return sents;
}

File: readability.py
# TODO
from cs50 import get_string


def main():
    # Get text from user
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sents(text)
    calc_CLI(letters, words, sentences)


# Function which counts letters from given text
def count_letters(text):
    letters = 0
    txt_len = len(text)
    for i in range(txt_len):
        if text[i] >= "A" and text[i] <= "Z":
            letters += 1
        if text[i] >= "a" and text[i] <= "z":
            letters += 1
    return letters


# Function which counts words from given text
def count_words(text):
    words = 1
    txt_len = len(text)
    for i in range(txt_len):
        if text[i] == " ":
            words += 1
    return words


# Function which counts sentences from given text
def count_sents(text):
    sents = 0
    txt_len = len(text)
    for i in range(txt_len):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sents += 1
    return sents


def calc_CLI(letters, words, sentences):
    # Define variables L and S for the Coleman-Liau index
    L = float(letters / words) * 100
    S = float(sentences / words) * 100
    # Coleman-Liau index formula
    CLI = round(0.0588 * L - 0.296 * S - 15.8)
    # Evaluating reading comprehension level based on the CLI value
    if CLI >= 16:
        print("Grade 16+")
        print()
    elif CLI >= 1:
        print(f"Grade {CLI}")
        print()
    else:
        print("Before Grade 1")
        print()


main()
